source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -37,7 +25,14 @@ FilterConfig::FilterConfig(UserMap&& users, const std::string& forward_username_
       authentication_header_(Http::LowerCaseString(authentication_header)),
       stats_(generateStats(stats_prefix + "basic_auth.", scope)) {}
 
-BasicAuthFilter::BasicAuthFilter(FilterConfigConstSharedPtr config) : config_(std::move(config)) {}
+BasicAuthFilter::BasicAuthFilter(FilterConfigConstSharedPtr config) : config_(std::move(config)) {
+  auto* factory = Envoy::Config::Utility::getFactoryByName<
+      Envoy::Extensions::Hash::NamedAlgorithmProviderConfigFactory>("envoy.hash.sha1");
+  if (factory == nullptr) {
+    throw EnvoyException("basic auth: did not find factory named 'envoy.hash.sha1'");
+  }
+  hash_algorithm_provider_ = factory->createAlgorithmProvider();
```

Comment:
I think it's acceptable to only support only hash algorithm for the BasicAuth filter, but we need to verify that the alorgithmProvider matches the configured Hash algorithm in the Htpasswd file when loading the passwords from the configuration.

Reply:
Done